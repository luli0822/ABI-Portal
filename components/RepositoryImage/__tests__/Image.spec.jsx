import React from 'react';
import { shallow } from 'enzyme';
import Image from '../index';

jest.mock('../../../images/responsive-images');

describe('<Image />', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<Image src="TestComponent/test.png" alt="alt-text" />);
		// https://github.com/facebook/jest/issues/7802
		// using .debug() instead of adding dependency on enzyme-to-json for now
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it('passes a className correctly', () => {
		const wrapper = shallow(
			<Image src="TestComponent/test.png" alt="alt-text" className="test-class-name" />
		);

		expect(wrapper.find('img.test-class-name').length).toBe(1);
	});
});
