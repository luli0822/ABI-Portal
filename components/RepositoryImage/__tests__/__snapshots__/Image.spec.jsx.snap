// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> renders correctly 1`] = `
"<picture>
  <Sources image={{...}} breakpointPixels={480} widthPercent={100} type=\\"image/png\\" isMinWidth={false} />
  <Sources image={{...}} breakpointPixels={767} widthPercent={100} type=\\"image/png\\" isMinWidth={false} />
  <Sources image={{...}} breakpointPixels={991} widthPercent={100} type=\\"image/png\\" isMinWidth={false} />
  <Sources image={{...}} breakpointPixels={1247} widthPercent={100} type=\\"image/png\\" isMinWidth={false} />
  <Sources image={{...}} breakpointPixels={1248} widthPercent={100} type=\\"image/png\\" isMinWidth={true} />
  <img src=\\"test-1440.png\\" className=\\"\\" alt=\\"alt-text\\" onLoad={[Function: onLoad]} />
</picture>"
`;
